src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -687,6 +756,57 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
             return allProperties;
         }
 
+        private static List<FieldInfo> GetAllFields(
+            [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
+            Type type)
+        {
+            var allFields = new List<FieldInfo>();
+
+            Type? baseType = type;
+            do
+            {
+                allFields.AddRange(baseType!.GetFields(DeclaredOnlyLookup));
+                baseType = baseType.BaseType;
+            }
+            while (baseType != typeof(object));
+
+            return allFields;
+        }
+
+        [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
+        private static object? BindParameter(ParameterInfo parameter, Type type, IConfiguration config,
+            BinderOptions options)
+        {
+            string? parameterName = parameter.Name;
+
+            if (parameterName is null)
+            {
+                throw new InvalidOperationException(SR.Format(SR.Error_ParameterBeingBoundToIsUnnamed, type));
+            }
+
+            var propertyBindingPoint = new BindingPoint(initialValue: config.GetSection(parameterName).Value, isReadOnly: false);
+
+            if (propertyBindingPoint.Value is null)
+            {
+                if (parameter.HasDefaultValue)
```

Comment:
We should be using the same method we use in DI to get the default value. There are a lot of nuances here (especially with supporting netstandard2.0)

[https://github.com/dotnet/runtime/blob/b098b6ffb9274fd9f9fd6db943e291ce9909af84/src/libraries/Common/src/Extensions/ParameterDefaultValue/ParameterDefaultValue.cs#L18-L20](https://github.com/dotnet/runtime/blob/b098b6ffb9274fd9f9fd6db943e291ce9909af84/src/libraries/Common/src/Extensions/ParameterDefaultValue/ParameterDefaultValue.cs#L18-L20)